### PR TITLE
- code change to return redirect http status for merged entries

### PR DESCRIPTION
**Pull request by LeonardoGonzales**

### CODE REVIEW

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java [hunk 1]
```diff
@@ -113,4 +126,18 @@ protected BasicSearchController(ApplicationEventPublisher eventPublisher,
 
     protected abstract String getEntityId(T entity);
 
+    protected abstract Optional<String> getEntityRedirectId(T entity);
+
```
Comment (eddturner) [inline common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java]: can remove extra new line

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java [hunk 2]
```diff
@@ -113,4 +126,18 @@ protected BasicSearchController(ApplicationEventPublisher eventPublisher,
...
+
+    private String getLocationURLForId(String redirectId) {
+        String path = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .build()
+                .getPath();
+        if (path != null) {
+            path = path.substring(0, path.lastIndexOf("/") + 1) + redirectId;
```
Comment (eddturner) [inline common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java]: sonarqube intellij plugin that using single quotes `'/'` in lastIndexOf will be faster than `"/"`.

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java [hunk 3]
```diff
@@ -49,13 +53,22 @@ protected BasicSearchController(ApplicationEventPublisher eventPublisher,
...
         context.setEntityOnly(true);
-
         if (contentType.equals(LIST_MEDIA_TYPE)) {
             context.setEntityIds(Stream.of(getEntityId(entity)));
         } else {
             context.setEntities(Stream.of(entity));
         }
-        return ResponseEntity.ok()
+
+        Optional<String> redirectId = getEntityRedirectId(entity);
```
Comment (eddturner) [inline common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java]: I think lines 62 - 70 can be replaced by:
```java
    ResponseEntity.BodyBuilder responseBuilder = getEntityRedirectId(entity)
                .map(id -> ResponseEntity.status(HttpStatus.SEE_OTHER)
                        .header(HttpHeaders.LOCATION, getLocationURLForId(id)))
                .orElse(ResponseEntity.ok());
```

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java (hunk 1 above)
Comment (LeonardoGonzales) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java]: Line removed... I also removed other extra lines in this class

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java (hunk 2 above)
Comment (LeonardoGonzales) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java]: Code changed to char. Good Catch..

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java (hunk 3 above)
Comment (LeonardoGonzales) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java]: I new you would comment on this.. LOL, but I never remember how to use it properly... tks!!!

**Pull request merged**